src/compiler/utilities.ts

```diff
@@ -6227,18 +6227,19 @@ namespace ts {
         };
     }
 
-    export function discoverProbableSymlinks(files: readonly SourceFile[], getCanonicalFileName: GetCanonicalFileName, cwd: string): SymlinkCache {
+    export function discoverProbableSymlinks(files: readonly SourceFile[], typeReferenceDirectives: readonly (ResolvedTypeReferenceDirective | undefined)[], getCanonicalFileName: GetCanonicalFileName, cwd: string): SymlinkCache {

```

Comment:
Can you write comment that we need to use typeReferenceDirectives here instead of from source file because of auto type reference directives for which there is no source file so those will be missed otherwise.